analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/ObjectCreation.cs

```diff
@@ -29,7 +30,13 @@ internal sealed class ObjectCreation : SimpleProcessor<IObjectCreationOperationW
 
     protected override ProgramState Process(SymbolicContext context, IObjectCreationOperationWrapper operation)
     {
-        if (operation.Type.IsNullableValueType())
+        if (operation.Type.IsAny(EmptyCollectionsShouldNotBeEnumeratedBase.TrackedCollectionTypes)
```

Comment:
`EmptyCollectionsShouldNotBeEnumeratedBase.TrackedCollectionTypes` is also used by the `Binary` processor.
Is this fine or should I move it?

Reply:
It should also be moved out of the rule.

Reply:
where should I put it?
What do we usually do for properties that >1 processors depend on?

Reply:
Let's do something similar to `ArithmeticCalculator`. Create an extra class and move the properties and relevant login inside (possibly also extract logic from `Binary`). Not sure about the name. Maybe `CollectionTracker`?